docs/BOOT_LOADER_INTERFACE.md

```diff
@@ -80,11 +80,20 @@ variables. All EFI variables use the vendor UUID
   * `1 << 5` → The boot loader supports looking for boot menu entries in the Extended Boot Loader Partition.
   * `1 << 6` → The boot loader supports passing a random seed to the OS.
 
-* The EFI variable `LoaderRandomSeed` contains a binary random seed if set. It
-  is set by the boot loader to pass an entropy seed read from the ESP to the OS.
-  The system manager then credits this seed to the kernel's entropy pool. It is
-  the responsibility of the boot loader to ensure the quality and integrity of
-  the random seed.
+* The EFI configuration table entry, `LINUX_EFI_RANDOM_SEED_TABLE_GUID`
+  (1ce1e5bc-7ceb-42f2-81e5-8aadf180f57b) is allocated with pool memory of type
+  `EfiACPIReclaimMemory`, and the boot loader uses it to pass an entropy seed
+  read from the ESP to the OS, which is read and then cleared by the kernel
+  before userspace starts. Its contents have the form:
+  ```
+  struct linux_efi_random_seed {
+      u32     size; // of the 'seed' array in bytes
+      u8      seed[];
+  };
+  ```
+  The size field is generally set to be 32 bytes, and the seed field includes a
+  hashed representation of any prior seed in `LINUX_EFI_RANDOM_SEED_TABLE_GUID`
+  together with the new seed.
```

Comment:
hmm, so BOOT_LOADER_INTERFACE is the specification how sd-boot communicates with userspace. i.e. this is where the variables and their meanings are originally defined, everything else is just an implementation of this spec.

But LINUX_EFI_RANDOM_SEED_TABLE_GUID is not really like that, it's defined by the Linux kernel, no? We just make use of the interface defined by the Linux kernel. Hence it makes sense to drop the variable, but the new documentaiton should probably be placed in some kernel docs somewhere?

(alternatively, since you wrote this, maybe add this as comment to the C sources?)

Reply:
Yea, the kernel side needs more docs, across several archs actually -- FDT, m68k, MIPS, and x86 have ways of passing seeds too. I'll work on that at some point.

I'll remove it from here, anyhow, and see if it fits in the RANDOM_SEEDS.md document or elsewhere.